test/e2e/serial/config/config.go

```diff
@@ -67,3 +79,77 @@ func GetRteCiImage() string {
 	}
 	return nropTestCIImage
 }
+
+func contains(items []string, st string) bool {
+	for _, item := range items {
+		if item == st {
+			return true
+		}
+	}
+	return false
+}
+
+var policyScopeToTmPolicy = map[string]nrtv1alpha2.TopologyManagerPolicy{
+	"restricted:pod":             nrtv1alpha2.RestrictedPodLevel,
+	"restricted:container":       nrtv1alpha2.RestrictedContainerLevel,
+	"best-effort:pod":            nrtv1alpha2.BestEffortPodLevel,
+	"best-effort:container":      nrtv1alpha2.BestEffortContainerLevel,
+	"none:":                      nrtv1alpha2.None,
+	"single-numa-node:pod":       nrtv1alpha2.SingleNUMANodePodLevel,
+	"single-numa-node:container": nrtv1alpha2.SingleNUMANodeContainerLevel,
+}
+
+func toTopolyPolicy(topologyManagerPolicy, topologyManagerScope string) (nrtv1alpha2.TopologyManagerPolicy, error) {
+	chain := topologyManagerPolicy + ":" + topologyManagerScope
+	tmPolicy, ok := policyScopeToTmPolicy[chain]
+	if !ok {
+		return "", fmt.Errorf("Unable to convert %q to TmPolicy", chain)
+	}
+	return tmPolicy, nil
+}
+
+func getTopologyConsistencyErrors(kconfigs map[string]*kubeletconfigv1beta1.KubeletConfiguration, nrts []nrtv1alpha2.NodeResourceTopology) map[string]error {
+	ret := make(map[string]error)
+	for nodeName, kconfig := range kconfigs {
+		nrt, err := e2enrt.FindFromList(nrts, nodeName)
+		if err != nil {
+			ret[nodeName] = fmt.Errorf("Unable to find KubeletConfig for node %q", nrt.Name)
+			continue
+		}
+
+		tmPolicy, err := toTopolyPolicy(kconfig.TopologyManagerPolicy, kconfig.TopologyManagerScope)
+		if err != nil {
+			ret[nodeName] = fmt.Errorf("Unable to convert kc.policy/kc.scope to TopologyMangerPolicy. node:%q, err: %w", nodeName, err)
+			continue
+		}
+
+		if !contains(nrt.TopologyPolicies, string(tmPolicy)) {
+			ret[nodeName] = fmt.Errorf("Incoherent KubeletConfig/NRT for node %q", nodeName)
+		}
+
+	}
+	return ret
+}
+
+func CheckNodesTopology(ctx context.Context) error {
```

Comment:
I believe we can avoid passing cxt to the function, and we can simply use context.TODO()

Reply:
We can, but we should not. We should not hardcode `context.TODO()` in the low-level function. The context should be created by the top-most function and passed through accordingly. TL;DR: I'd keep the context here and in all the utility functions.

Reply:
ah good to learn about that, thanks for sharing it.